org.eclipse.xtext.xbase.testing/src/org/eclipse/xtext/xbase/testing/CompilationTestHelper.java

```diff
@@ -147,6 +150,17 @@ public void setJavaCompilerClassPath(ClassLoader classLoader) {
 		this.javaCompiler = new OnTheFlyJavaCompiler2(classLoader, generatorConfigProvider.get(null).getJavaSourceVersion());
 		this.classpathUriContext = classLoader;
 	}
+
+	/**
+	 * Sets the Java version both for the DSL generator, for example, Xbase
+	 * compiler, and for the Java compiler.
+	 * 
+	 * @since 2.11
+	 */
+	public void setJavaVersion(JavaVersion javaVersion) {
+		this.javaCompiler.setJavaVersion(javaVersion);
+		generatorConfigProvider.get(null).setJavaSourceVersion(javaVersion);
```

Comment:
@svenefftinge that looks odd to me as well, but I just mimicked what is already being done in the existing code: https://github.com/eclipse/xtext-extras/blob/7e358229d6f5cee004ab9cb0389af49b95a230ed/org.eclipse.xtext.xbase.testing/src/org/eclipse/xtext/xbase/testing/CompilationTestHelper.java#L150